scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/IncrementalSimulationDriver.java

```diff
@@ -0,0 +1,247 @@
+package gov.nasa.jpl.aerie.scheduler;
+
+import gov.nasa.jpl.aerie.merlin.driver.MissionModel;
+import gov.nasa.jpl.aerie.merlin.driver.SerializedActivity;
+import gov.nasa.jpl.aerie.merlin.driver.SimulationResults;
+import gov.nasa.jpl.aerie.merlin.driver.engine.SimulationEngine;
+import gov.nasa.jpl.aerie.merlin.driver.timeline.LiveCells;
+import gov.nasa.jpl.aerie.merlin.driver.timeline.TemporalEventSource;
+import gov.nasa.jpl.aerie.merlin.protocol.driver.Scheduler;
+import gov.nasa.jpl.aerie.merlin.protocol.model.Task;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.TaskStatus;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+import org.jetbrains.annotations.NotNull;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IncrementalSimulationDriver {
+
+  private Duration curTime;
+  private SimulationEngine engine;
+  private LiveCells cells;
+  private TemporalEventSource timeline;
+  private MissionModel<?> missionModel;
+
+  //mapping each activity name to its task id (in String form) in the simulation engine
+  private Map<String, String> actToId;
+
+  //boolean stating whether the simulation results should be regenerated
+  private boolean areLastSimResultsDirty;
+
+  //simulation results so far
+  private SimulationResults lastSimResults;
+  //cached simulation results cover the period [Duration.ZERO, lastSimResultsEnd]
+  private Duration lastSimResultsEnd = Duration.ZERO;
+
+  //map from activity start time to serialized activity and its name
+  private List<SimulatedActivity> activitiesInserted = new ArrayList<>();
```

Comment:
This comment is inaccurate now that this is a `List`. Aside from that, this comment mostly repeats what the code says (or had said, as a `Map`). It may be helpful to add more semantic information, e.g.:
```suggestion
  // The collection of activities scheduled so far.
  private List<SimulatedActivity> activitiesInserted = new ArrayList<>();
```